scripts/container-only/wkdev-setup-vscode

```diff
@@ -96,15 +133,78 @@ install_extensions() {
     install_extension ms-python.python "Python support" true
 }
 
+# These are VERY helpful for WebKit development, but we won't override existing settings if the user already has them.
+default_settings() {
+    if argsparse_is_option_set "proprietary"; then
+        VSCODE_CONFIG_PATH=${HOME}/.config/Code/User/
+    else
+        VSCODE_CONFIG_PATH=${HOME}/.config/VSCodium/User/
+    fi
+    if [[ ! -e "${VSCODE_CONFIG_PATH}/settings.json" ]]; then
+        mkdir -p "${VSCODE_CONFIG_PATH}"
+        tee "${VSCODE_CONFIG_PATH}/settings.json" << HERE
+{
+    "clangd.arguments": [
+        "-header-insertion=never"
+    ],
+    "terminal.integrated.scrollback": 1000000000,
+    "editor.renderWhitespace": "trailing",
+    "window.menuBarVisibility": "toggle",
+    "workbench.colorCustomizations": {
+    "editorWhitespace.foreground": "#FF0000",
+    "editorWhitespace.background": "#FF0000"
+},
+"terminal.integrated.allowChords": false,
+}
```

Comment:
Some of these are pretty arbitrary. I like the `clangd.arguments` but maybe lets move this to a different PR to discuss.

Reply:
The clangd and fg/bg colours are important for our style guidelines, but I removed the rest.

Reply:
I'm not sure I know what guidelines you refer to

Reply:
Oh you just mean to more easily visualize whitespace.

I've never needed that and don't think I'd want it by default. I'll check how it looks.

Reply:
I'm going to vote no on this, I think it would annoy most people.

![image](https://github.com/user-attachments/assets/ceadf1b0-d812-4a81-a00a-58a5c50b32f5)

Reply:
Oh shoot, it doesn't look like that for me. It is only supposed to display trailing whitespace

Reply:
![2025-01-27T13:27:24,752795531-07:00](https://github.com/user-attachments/assets/c2ad6085-4799-41fa-b8a2-1f1360b303f9)